src/Dropdown/Dropdown.js

```diff
@@ -2,57 +2,185 @@ import PropTypes from 'prop-types';
 import React, { Component } from 'react';
 import onClickOutside from 'react-onclickoutside';
 import classNames from 'classnames';
+import Option from './Option';
+import sameValue from '../utils/sameValue';
+import debounce from '../utils/debounce';
 
+const DOWN = -1;
+const EMPTY_STRING = '';
+const INITIAL_INDEX = -1;
+const KEYS = [13, 27, 38, 40, 32];
 const LEFT = 'Left';
 const RIGHT = 'Right';
+const UP = 1;
+const WAIT_TIME = 200;
+const [ENTER, ESC, ARROW_UP, ARROW_DOWN, SPACE] = KEYS;
+const getA11yStatusMessage = ({
+  isOpen,
+  options = [],
+  selectedOption = {}
+}) => {
+  const isClosed = !isOpen;
+  const { label: selectedLabel } = selectedOption;
+
+  if (isClosed) {
+    return `${selectedLabel} is selected.`;
+  }
+
+  const resultCount = options.length;
+
+  return `${resultCount} ${
+    resultCount === 1 ? 'option is' : 'options are'
+  } available, use up and down arrow keys to navigate. Press Enter key to select or Escape key to cancel.`;
+};
 
 export class Dropdown extends Component {
   constructor(props) {
     super(props);
     this.state = {
+      a11yStatusMessage: EMPTY_STRING,
+      selectedIndex: INITIAL_INDEX,
       selectedValue: this.props.defaultValue,
       isOpen: false,
       upward: this.props.upward,
       lineUp: RIGHT
     };
+
+    this.activatorRef = React.createRef();
     this.dropdownRef = React.createRef();
     this.optionsRef = React.createRef();
   }
 
-  getSelectedLabel() {
+  componentDidMount() {
+    this.updateA11yMessage();
+  }
+
+  get optionsExceptSelected() {
     const { options } = this.props;
     const { selectedValue } = this.state;
-    const selectedOption = options.find(option => {
-      return this.sameValue(option.value, selectedValue);
-    });
+    const notSelected = option => !sameValue(option.value, selectedValue);
```

Comment:
Why are we using sameValue instead of comparing with != ?

Reply:
`sameValue` also converts the strings to uppercase, so it has more logic behind...

Reply:
Ok